devcycle_python_sdk/models/event.py

```diff
@@ -18,22 +18,26 @@ class EventType:
 class Event:
     type: Optional[str] = None
     target: Optional[str] = None
-    date: int = field(default_factory=lambda: int(time.time()))
+    date: datetime = field(default_factory=lambda: datetime.utcnow())
     value: Optional[int] = None
-    metaData: Dict[str, str] = field(default_factory=dict)
+    metaData: Optional[Dict[str, str]] = None
 
     def to_json(self):
-        return {
+        json_dict = {
             key: getattr(self, key)
             for key in self.__dataclass_fields__
-            if getattr(self, key) is not None
+            if getattr(self, key) is not None and key != "date"
         }
+        if self.date:
+            json_dict["date"] = f"{self.date.isoformat()}Z"
```

Comment:
This date field is user-provided, right? Is there a risk here that they provide us a non-UTC datetime, and we ignore the timezone and use UTC?

Reply:
hmm, fair point.  i didn't get a straight answer out of adam as to what should be used for timezone.  Maybe the right thing to do is convert it to UTC and then format it?

Reply:
Probably, yeah. We might need the `pytz` library for that, from what I remember `datetime` just provides the attribute for a timezone but not the actual values?

Unless we can define one for UTC ourselves?

Reply:
Just read up on it some more - I think as long as we only want to support converting to UTC and not other more exotic timezones, we can do something like this:
```
from datetime import datetime, timezone, timedelta
...
dt.astimezone(timezone.utc)
```
And then if we receive a `datetime` with a timezone defined by some other library (pytz, etc) we'll still be able to convert it.

Reply:
ok, i added a conversion - just force it before changing it to json